### PR TITLE
feat: pretty print for InternalServerError

### LINKED ISSUE #1399: Improve pretty print for InternalServerError component
When the application raise an exception usually the error is catch and handled by  `InternalServerError`  component. 
UI unit test framework helps investigating the cause of the error by printing the current UI component tree, with sensible details for each component.
On the browser `InternalServerError` component shows the error message, but for the UI tests the information is not shown.

```
java.lang.AssertionError: An internal server error occurred; please check log for the actual stack-trace. Error text: 
└── UI[]
    └── InternalServerError[]
```

It would be helpful to have the UI pretty print also dump the `InternalServerError` error message.

### PR DESCRIPTION
**Pull request by mcollovati**
Screenshot will print target view name, error message and stacktrace

Fixes #1399

### CODE REVIEW

File: vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt [hunk 1]
```diff
@@ -150,6 +151,17 @@ fun Component.toPrettyString(): String {
...
+        println(this.element.outerHTML)
+        val view = this.element.getChild(0).outerHTML.replace(Regex("(?s).* trying to navigate to '([^']+).*"), "$1")
+        val errorMessage = this.element.getChild(0).outerHTML.
+            replace(Regex("(?s).* with the (?:root cause|exception message) '([^']+).*"), "$1")
+        val stackTrace: String = this.element.outerHTML.trim().substringAfter("<pre>").substringBefore("</pre>")
+        //list.add(outerHtml.ellipsize(100))
+        list.add("targetView='$view'")
+        list.add("error='$errorMessage'")
+        list.add("stacktrace='${stackTrace.substringAfter("\n")}'")
+    }
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt:164]: Perhaps we should just create test error page extending InternalServerError that would store the data into parameters.
Then they would be added automatically.

File: vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt [hunk 2]
```diff
@@ -150,6 +151,17 @@ fun Component.toPrettyString(): String {
...
     }
+    if (this is InternalServerError) {
+        println(this.element.outerHTML)
+        val view = this.element.getChild(0).outerHTML.replace(Regex("(?s).* trying to navigate to '([^']+).*"), "$1")
+        val errorMessage = this.element.getChild(0).outerHTML.
+            replace(Regex("(?s).* with the (?:root cause|exception message) '([^']+).*"), "$1")
+        val stackTrace: String = this.element.outerHTML.trim().substringAfter("<pre>").substringBefore("</pre>")
+        //list.add(outerHtml.ellipsize(100))
+        list.add("targetView='$view'")
+        list.add("error='$errorMessage'")
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt:162]: errorMessage might be better than error.

Comment (mcollovati) [reply vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt:162]: `errorMessage` is suppressed by pretty printer. I'll use `failureMessage`

File: vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt [hunk 3]
```diff
@@ -21,6 +21,7 @@ import com.vaadin.flow.component.html.Anchor
 import com.vaadin.flow.component.icon.Icon
 import com.vaadin.testbench.unit.internal.PrettyPrintTree.Companion.ofVaadin
 import kotlin.reflect.jvm.isAccessible
+import com.vaadin.flow.router.InternalServerError
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/PrettyPrintTree.kt]: done

**Review by caalador — APPROVED** [review]

**Pull request merged**